src/core_plugins/elasticsearch/lib/map_uri.js

```diff
@@ -14,7 +15,8 @@ module.exports = function mapUri(server, prefix) {
     }
     const query = querystring.stringify(request.query);
     if (query) url += '?' + query;
-    const filteredHeaders = filterHeaders(request.headers, server.config().get('elasticsearch.requestHeadersWhitelist'));
-    done(null, url, filteredHeaders);
+    const filteredHeaders = filterHeaders(request.headers, config.get('elasticsearch.requestHeadersWhitelist'));
+    const customHeaders = setHeaders(filteredHeaders, config.get('elasticsearch.customHeaders'));
```

Comment:
Mostly curious: what was your rationale behind creating `setHeaders` as opposed to using something like [`_.merge`](https://lodash.com/docs#merge) here?